### PR TITLE
API to get SHA256 of a model file

### PR DESCRIPTION
**Pull request by JeLuF**
To be used from javascript to collect metadata from civitai 
https://civitai.com/api/v1/model-versions/by-hash/0A35347528

### CODE REVIEW

File: ui/easydiffusion/server.py [hunk 1]
```diff
@@ -451,3 +455,26 @@ def modify_package_internal(package_name: str, req: dict):
         log.error(str(e))
         log.error(traceback.format_exc())
         return HTTPException(status_code=500, detail=str(e))
+
+def get_sha256_internal(obj_path):
+    import hashlib
+    from easydiffusion.utils import sha256sum
+
+    path = obj_path.split("/")
```
Comment (cmdr2) [inline ui/easydiffusion/server.py:463]: Do we need to check for backslash as well? I've noticed that the model paths with nested subfolders can contain them.

Comment (cmdr2) [reply ui/easydiffusion/server.py:463]: Or we could ensure that the JS side never sends backslashes

**Comment by cmdr2** [comment]
Thanks! This will be a very handy feature

File: ui/easydiffusion/server.py (hunk 1 above)
Comment (JeLuF) [reply ui/easydiffusion/server.py:463]: /get/models returns a JSON with nested objects, without any separator. So it's the javascript side that needs to make this into a path.

**Pull request merged**